app/components/primer/auto_complete.rb

```diff
@@ -1,88 +1,115 @@
 # frozen_string_literal: true
 
 module Primer
-  # Use `AutoComplete` to populate input values from server search results.
+  # Use `AutoComplete` to provide a user with a list of selectable suggestions that appear when they type into the
+  # input field. This list is populated by server search results.
   # @accessibility
-  #   Always provide a label for the `Autocomplete` component.
+  #   Always provide an accessible label to help the user interact with the input element and listbox popup.
+  #
+  #   To show a visible label, set the `label` slot. The`for` attribute must be set to the `id` of
+  #   `input` in order for the `<label>` to be properly linked.
+  #
+  #   If you do not wish to provide a visible label, you must set an `aria-label` attribute. You may set
+  #   `:"aria-label"` directly on `AutoComplete` instead of the slots and Primer will apply it to the correct elements.
   class AutoComplete < Primer::Component
     status :beta
 
     DEFAULT_INPUT_TYPE = :text
     INPUT_TYPE_OPTIONS = [DEFAULT_INPUT_TYPE, :search].freeze
 
+    # Optionally render a visible label. See <%= link_to_accessibility %>
+    #
+    # @param for [Symbol] id of input
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :label, lambda { |**system_arguments|
+      raise ArgumentError, "missing keyword: for" if !system_arguments.key?(:for) && !Rails.env.production?
+
+      @input_id_for_label = system_arguments[:for]
```

Comment:
I think this may be relying on the order in which slots are called, I'd prefer if we could set an `input_id:` argument in the parent that will pass it directly to `label` and `input`